pkg/task/task.go

```diff
@@ -0,0 +1,61 @@
+package task
+
+import "time"
+
+const (
+	TaskStateRequested TaskState = iota // when client requests a task
+	TaskStateScheduled                  // daemon accepts the task into the queue
+	TaskStateBuilding                   // daemon is building the task
+	TaskStateRunning                    // daemon is running the task
+	TaskStateComplete                   // daemon is not running the task. ready to collect.
```

Comment:
These are under the `task` package, you can drop the `Task` or even `TaskState` prefix.

`TaskState` -> `State`.

Reply:
As commented on Slack, I think it'd be cleaner if we have the following states:
1. `requested` (added to the queue, but not processed)
2. `in progress / processing` (popped from the queue, and building or running or both, depending on the request)
3. `completed` (no longer running (could be successful or failed or any other terminal state) ; outputs are ready to be processed; note that at the moment we don't have history on `failed` tasks (for example is a pod goes OOM and we don't have `keep_alive`, we won't find out - we need to think about that).

---

I think these 3 are enough, not sure why we need 5 states.

Reply:
Done!